library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileOutputOperator.java

```diff
@@ -484,34 +484,40 @@ public FSFilterStreamContext load(@Nonnull String filename)
         //of this operator.
         for(String seenFileName: endOffsets.keySet()) {
           try {
-            Integer part = openPart.get(seenFileName).getValue() + 1;
+            Integer fileOpenPart = this.openPart.get(seenFileName).getValue();
+            int nextPart = fileOpenPart + 1;
```

Comment:
One more thing on line 446 there is an optimization to not create the new tmp file if the file hasn't changed from the recovery offset (i.e., file.length == recovery offset). This may be a problem as some other process holds the write lease to this file and we may not be able to open it in append mode later on when data is going to be written